WordPress/Classes/ViewRelated/Blog/Site Settings/HomepageSettingsViewController.swift

```diff
@@ -96,59 +105,141 @@ import WordPressShared
 
         return [
             CheckmarkRow(title: HomepageType.posts.title, checked: homepageType == .posts, action: { [weak self] _ in
-                self?.setHomepageType(.posts)
+                if self?.inProgressChange == nil {
+                    self?.update(with: .homepageType(.posts))
+                }
             }),
             CheckmarkRow(title: HomepageType.page.title, checked: homepageType == .page, action: { [weak self] _ in
-                self?.setHomepageType(.page)
+                if self?.inProgressChange == nil {
+                    self?.update(with: .homepageType(.page))
+                }
             })
         ]
     }
 
     var selectedPagesRows: [ImmuTableRow] {
-        var homepageTitle = ""
-        var postsPageTitle = ""
+        let homepageID = blog.homepagePageID
+        let homepage = homepageID.flatMap { postService.findPost(withID: NSNumber(value: $0), in: blog) }
+        let homepageTitle = homepage?.titleForDisplay() ?? ""
+
+        let postsPageID = blog.homepagePostsPageID
+        let postsPage = postsPageID.flatMap { postService.findPost(withID: NSNumber(value: $0), in: blog) }
+        let postsPageTitle = postsPage?.titleForDisplay() ?? ""
+
+        let homepageRow = pageSelectionRow(title: Strings.homepage,
+                                                      detail: homepageTitle,
+                                                      isInProgress: HomepageChange.isSelectedHomepage,
+                                                      changeForPost: { .selectedHomepage($0) })
+        let postsPageRow = pageSelectionRow(title: Strings.postsPage,
+                                                       detail: postsPageTitle,
+                                                       isInProgress: HomepageChange.isSelectedPostsPage,
+                                                       changeForPost: { .selectedPostsPage($0) })
+        return [homepageRow, postsPageRow]
+    }
 
-        if let homepageID = blog.homepagePageID,
-            let homepage = postService.findPost(withID: NSNumber(value: homepageID), in: blog) {
-            homepageTitle = homepage.titleForDisplay()
+    private func pageSelectionRow(title: String, detail: String, isInProgress: (HomepageChange) -> Bool, changeForPost: @escaping (Int) -> HomepageChange) -> ImmuTableRow {
+        if let inProgressChange = inProgressChange, isInProgress(inProgressChange) {
+            return ActivityIndicatorRow(title: title, animating: true, action: nil)
+        } else {
+            return NavigationItemRow(title: title, detail: detail, action: { [weak self] _ in
+                self?.showPageSelection(selectedPostID: self?.blog?.homepagePageID, hiddenPostID: self?.blog?.homepagePostsPageID, change: changeForPost)
```

Comment:
We're always passing homepageID for the selected post, and postPageID for the hidden post ID here, so the picker shown the same content no matter which row you're tapping into. These should be switched for the Posts row.